sklearn/inspection/_plot/partial_dependence.py

```diff
@@ -775,6 +778,57 @@ def _get_sample_count(self, n_samples):
             return ceil(n_samples * self.subsample)
         return n_samples
 
+    def _get_color_dict_from_values(
+        self, value_list, categorical_or_continuous: str, cmap_name: str = "Spectral"
+    ):
+        color_dict = {}
+        if categorical_or_continuous == "categorical":
+            mapping = {
+                np.unique(value_list)[i]: f"C{i}"
```

Comment:
da du hier sowohl die werte als auch den index willst, benutz am besten enumerate:
`{v: f"C{i}" for i, v in enumerate(np.unique(value_list))}`